api/agent/drivers/docker/docker_client.go

```diff
@@ -263,6 +264,12 @@ func (d *dockerWrap) retry(ctx context.Context, f func() error) error {
 		default:
 		}
 
+		if isSyslogError(err) {
+			// syslog error is an user error
+			e := models.NewAPIError(500, errors.New("Internal server error (SYSLOG)"))
```

Comment:
Minor nit, but I think it would be good to hoist this to models/error.go, see ferr{} class of errors there... (eg. ErrRequestContentTooBig as example.) Maybe we could clarify the intent too, "failed to connect to syslog" or something like that perhaps?

Reply:
Yeah, the issue is that it is the function consumer who is getting the error, not the function owner.